File: python-version/LC36/is_valid_sudoku.py
from typing import List


class Solution:
    @staticmethod
    def isValidSudoku(board: List[List[str]]) -> bool:
        row, col = [[False] * 9 for _ in range(9)], [[False] * 9 for _ in range(9)]
        box = [[[False] * 9 for _ in range(3)] for _ in range(3)]
        for i in range(9):
            for j in range(9):
                if board[i][j] == '.':
                    continue
                num = ord(board[i][j]) - ord('1')
                if row[i][num] or col[j][num] or box[i//3][j//3][num]:
                    return False
                row[i][num] = True
                col[j][num] = True
                box[i//3][j//3][num] = True
        return True

File: go-version/leetcode/LC36/is_valid_sudoku.go
package LC36

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/8/30 上午11:31
 * @FilePath: is_valid_sudoku
 * @Description: 判断有效数独
 */

func isValidSudoku(board [][]byte) bool {
	var row, col [9][9]bool
	var box [3][3][9]bool
	for i := 0; i < 9; i++ {
		for j := 0; j < 9; j++ {
			if board[i][j] == '.' {
				continue
			}
			num := board[i][j] - '1'
			if row[i][num] || col[j][num] || box[i/3][j/3][num] {
				return false
			}
			row[i][num] = true
			col[j][num] = true
			box[i/3][j/3][num] = true
		}
	}
	return true
}
